bot/cogs/snakes.py

```diff
@@ -0,0 +1,1208 @@
+
+            snake_info["info"] = info
+
+        return snake_info
+
+    async def _get_snake_name(self) -> Dict[str, str]:
+        """
+        Gets a random snake name.
+        :return: A random snake name, as a string.
+        """
+
+        response = await self.bot.http_session.get(self.names_url, headers=self.headers)
+        name_data = await response.json()
+
+        return name_data
+
+    async def _validate_answer(self, ctx: Context, message: Message, answer: str, options: list):
+        """
+        Validate the answer using a reaction event loop
+        :return:
+        """
+
+        def predicate(reaction, user):
+            """
+            Test if the the answer is valid and can be evaluated.
+            """
+            return (
+                reaction.message.id == message.id                  # The reaction is attached to the question we asked.
+                and user == ctx.author                             # It's the user who triggered the quiz.
+                and str(reaction.emoji) in ANSWERS_EMOJI.values()  # The reaction is one of the options.
+            )
+
+        for emoji in ANSWERS_EMOJI.values():
+            await message.add_reaction(emoji)
+
+        # Validate the answer
+        try:
+            reaction, user = await ctx.bot.wait_for("reaction_add", timeout=45.0, check=predicate)
+        except asyncio.TimeoutError:
+            await ctx.channel.send(f"You took too long. The correct answer was **{options[answer]}**.")
+            await message.clear_reactions()
+            return
+
+        if str(reaction.emoji) == ANSWERS_EMOJI[answer]:
+            await ctx.send(f"{random.choice(CORRECT_GUESS)} The correct answer was **{options[answer]}**.")
+        else:
+            await ctx.send(
+                f"{random.choice(INCORRECT_GUESS)} The correct answer was **{options[answer]}**."
+            )
+
+        await message.clear_reactions()
+    # endregion
+
+    # region: Commands
+    @bot_has_permissions(manage_messages=True)
+    @command(name="snakes.antidote()", aliases=["snakes.antidote"])
+    async def antidote(self, ctx: Context):
+        """
+        Antidote - Can you create the antivenom before the patient dies?
+
+        Rules:  You have 4 ingredients for each antidote, you only have 10 attempts
+                Once you synthesize the antidote, you will be presented with 4 markers
+                Tick: This means you have a CORRECT ingredient in the CORRECT position
+                Circle: This means you have a CORRECT ingredient in the WRONG position
+                Cross: This means you have a WRONG ingredient in the WRONG position
+
+        Info:   The game automatically ends after 5 minutes inactivity.
+                You should only use each ingredient once.
+
+        This game was created by Lord Bisk and Runew0lf.
+        """
+
+        def event_check(reaction_: Reaction, user_: Member):
```

Comment:
Seems like this should probably be just a lambda or moved out of the function into the class.

Reply:
:ok_hand: